localstack/services/cloudformation/engine/template_preparer.py

```diff
@@ -29,21 +37,156 @@ def template_to_json(template: str) -> str:
+        "requestId": long_uid(),
+        "templateParameterValues": formatted_stack_parameters,
+    }
+
+    client = aws_stack.connect_to_service("lambda")
+    invocation = client.invoke(
+        FunctionName=macro_definition["FunctionName"], Payload=json.dumps(event)
+    )
+    if invocation.get("StatusCode") != 200 or invocation.get("FunctionError") == "Unhandled":
+        raise FailedTransformation(
+            transformation=macro["Name"],
+            message=f"Received malformed response from transform {transformation_id}. Rollback requested by user.",
+        )
+    result = json.loads(invocation["Payload"].read())
+
+    if result.get("status") != "success":
+        error_message = result.get("errorMessage")
+        message = (
+            f"Transform {transformation_id} failed with: {error_message}. Rollback requested by user."
+            if error_message
+            else f"Transform {transformation_id} failed without an error message.. Rollback requested by user."
+        )
+        raise FailedTransformation(transformation=macro["Name"], message=message)
+
+    if not isinstance(result.get("fragment"), dict):
+        raise FailedTransformation(
+            transformation=macro["Name"],
+            message="Template format error: unsupported structure.. Rollback requested by user.",
+        )
+
+    return result.get("fragment")
+
+
+def apply_serverless_transformation(parsed_template):
     """only returns string when parsing SAM template, otherwise None"""
-    # template_body = get_template_body(req_data) # FIXME
-    parsed = parse_template(template_body)
-    if parsed.get("Transform") == "AWS::Serverless-2016-10-31":
+    region_before = os.environ.get("AWS_DEFAULT_REGION")
+    if boto3.session.Session().region_name is None:
+        os.environ["AWS_DEFAULT_REGION"] = aws_stack.get_region()
+    loader = create_policy_loader()
 
+    try:
+        transformed = transform_sam(parsed_template, {}, loader)
+        return transformed
+    except Exception as e:
+        raise FailedTransformation(transformation=SERVERLESS_TRANSFORM, message=str(e))
+    finally:
         # Note: we need to fix boto3 region, otherwise AWS SAM transformer fails
-        region_before = os.environ.get("AWS_DEFAULT_REGION")
-        if boto3.session.Session().region_name is None:
-            os.environ["AWS_DEFAULT_REGION"] = aws_stack.get_region()
-        loader = create_policy_loader()
-        try:
-            transformed = transform_sam(parsed, {}, loader)
-            return json.dumps(transformed)
-        finally:
-            os.environ.pop("AWS_DEFAULT_REGION", None)
-            if region_before is not None:
-                os.environ["AWS_DEFAULT_REGION"] = region_before
+        os.environ.pop("AWS_DEFAULT_REGION", None)
+        if region_before is not None:
+            os.environ["AWS_DEFAULT_REGION"] = region_before
+
+
+def format_transforms(transforms: List | Dict | str) -> List[Dict]:
```

Comment:
`List`, `Dict`, etc. should all be lower cased. The upper case variants where only required in earlier python versions.

Reply:
nit: would be great to have  a short docstring here explaining the context of the util